src/python/tests/annotation/test_update_description.py

```diff
@@ -0,0 +1,307 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.annotation.load` module.
+"""
+
+from contextlib import nullcontext as no_raise
+from pathlib import Path
+from typing import ContextManager
+from sqlalchemy import text
+
+import pytest
+from pytest import CaptureFixture, param, raises
+
+from ensembl.io.genomio.annotation import get_core_data, load_descriptions
+from ensembl.core.models import Gene, Transcript, ObjectXref, Xref, metadata
+from ensembl.utils.database import UnitTestDB
+
+
+def add_gene(dialect: str, session, gene_data=dict) -> None:
+    """Add a gene and a transcript from a gene_data dict.
+
+    Args:
+    dialect: if "mysql", allow to skip foreign key check.
+    session: SQLalchemy session.
+    gene_data:
+        "gene_name" -> gene.stable_id
+        "gene_desc" -> gene.description
+        "tr_name" -> transcript.stable_id
+        "tr_desc" -> transcript.description
+        "gene_xref" -> xref display_name attached to the gene
+    """
+    gene_name = gene_data.get("gene_name", "gene1")
+    gene_description = gene_data.get("gene_desc", "")
+    gene_xref = gene_data.get("gene_xref", "")
+    tr_name = gene_data.get("tr_name", "tr1")
+    tr_description = gene_data.get("tr_desc", "")
+    if dialect == "mysql":
+        session.execute(text("SET FOREIGN_KEY_CHECKS=0"))
```

Comment:
I was hoping this wasn't necessary, because it brings very infrastructure specific elements into a high-level abstraction layer :cry: Maybe this is something we may need to reconsider in the schema.

Reply:
Can't help it, otherwise we'd need to add base test data for the whole database (we might want to do that at some point, but at least here we keep control of everything in a limited manner)